samples/user-interface/windowmanager/src/main/java/com/example/platform/ui/windowmanager/MediaPlayerActivity.kt

```diff
@@ -0,0 +1,162 @@
+        super.onCreate(savedInstanceState)
+
+        binding = ActivityMediaPlayerBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        lifecycleScope.launch(Dispatchers.Main) {
+            lifecycle.repeatOnLifecycle(Lifecycle.State.STARTED) {
+                WindowInfoTracker.getOrCreate(this@MediaPlayerActivity)
+                    .windowLayoutInfo(this@MediaPlayerActivity)
+                    .collect { newLayoutInfo ->
+                        onLayoutInfoChanged(newLayoutInfo)
+                    }
+            }
+        }
+    }
+
+    override fun onStart() {
+        super.onStart()
+        initializePlayer()
+    }
+
+    override fun onResume() {
+        super.onResume()
+        hideSystemUI()
+    }
+
+    override fun onStop() {
+        super.onStop()
+        releasePlayer()
+    }
+
+    private fun initializePlayer() {
+        if (player == null) {
+            val trackSelector = DefaultTrackSelector(this)
+            trackSelector.setParameters(
+                trackSelector.buildUponParameters().setMaxVideoSizeSd()
+            )
+            player = ExoPlayer.Builder(this)
+                .setTrackSelector(trackSelector)
+                .build()
+        }
+        binding.playerView.player = player
+        val mediaItem = MediaItem.fromUri(getString(R.string.media_url_mp4))
+        player?.setMediaItem(mediaItem)
+        player?.playWhenReady = playWhenReady
+        player?.seekTo(currentMediaIndex, playbackPosition)
+        player?.prepare()
+        binding.controlView.player = player
+    }
+
+    private fun releasePlayer() {
+        player?.let { player ->
+            playbackPosition = player.currentPosition
+            currentMediaIndex = player.currentMediaItemIndex
+            playWhenReady = player.playWhenReady
+            player.release()
+        }
+        player = null
+    }
+
+    @Suppress("DEPRECATION")
+    private fun hideSystemUI() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+            window.insetsController?.let {
+                it.systemBarsBehavior = WindowInsetsController.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
+                it.hide(WindowInsets.Type.systemBars())
+            }
+        } else {
+            window.decorView.systemUiVisibility = (
+                    View.SYSTEM_UI_FLAG_IMMERSIVE
+                            or View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
+                            or View.SYSTEM_UI_FLAG_FULLSCREEN
+                            or View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
+                            or View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
+                    )
+            window.addFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_NAVIGATION)
+        }
+    }
+
+    private fun centerPlayer() {
+        binding.playerView.useController = true // use embedded controls
+        ConstraintLayout.getSharedValues().fireNewValue(R.id.fold, 0)
+    }
+
+    private fun foldPlayer(fold: Int) {
```

Comment:
I would add a comment that specifies that when the phone is in tabletop mode, the player won't use embedded controller but rather the external one declared in the manifest

Reply:
Done, thanks!